junit-jupiter-api/src/main/java/org/junit/jupiter/api/parallel/ResourceLocksProvider.java

```diff
@@ -0,0 +1,223 @@
+	 */
+	default Set<Lock> provideForClass(Class<?> testClass) {
+		return emptySet();
+	}
+
+	/**
+	 * Add shared resources to a {@linkplain Nested nested} test class.
+	 *
+	 * <p>Invoked in case:
+	 * <ul>
+	 *     <li>an enclosing test class of any level or its parent class
+	 *     is annotated with {@code @ResourceLock(providers)}.</li>
+	 *     <li>a nested test class or its parent class
+	 * 	   is annotated with {@code @ResourceLock(providers)}.</li>
+	 * </ul>
+	 *
+	 * <p>Invoked for a nested test class
+	 * annotated with {@code @ResourceLock(providers)}
+	 * and for its child classes.
+	 *
+	 * @apiNote Adding {@linkplain Lock a shared resource} with this method
+	 * has the same semantics as annotating a nested test class
+	 * with analogous {@code @ResourceLock(value, mode)}.
+	 *
+	 * @param testClass a nested test class to add shared resources
+	 * @return a set of {@link Lock}; may be empty
+	 * @see Nested
+	 */
+	default Set<Lock> provideForNestedClass(Class<?> testClass) {
+		return emptySet();
+	}
+
+	/**
+	 * Add shared resources to a test method.
+	 *
+	 * <p>Invoked in case:
+	 * <ul>
+	 *     <li>an enclosing test class of any level or its parent class
+	 *     is annotated with {@code @ResourceLock(providers)}.</li>
+	 *     <li>a test method
+	 *     is annotated with {@code @ResourceLock(providers)}.</li>
+	 * </ul>
+	 *
+	 * @apiNote Adding {@linkplain Lock a shared resource} with this method
+	 * has the same semantics as annotating a test method
+	 * with analogous {@code @ResourceLock(value, mode)}.
+	 *
+	 * @param testClass a test class
+	 * or {@linkplain Nested nested} test class containing the {@code testMethod}
+	 * @param testMethod a test method to add shared resources
+	 * @return a set of {@link Lock}; may be empty
+	 */
+	default Set<Lock> provideForMethod(Class<?> testClass, Method testMethod) {
+		return emptySet();
+	}
+
+	/**
+	 *
+	 * <p>{@link Lock} represents a shared resource.
+	 *
+	 * <p>Each resource is identified by {@link #key}.
+	 * In addition,{@link #accessMode} allows you to specify
+	 * whether a test class or test
+	 * method requires {@link ResourceAccessMode#READ_WRITE READ_WRITE}
+	 * or only {@link ResourceAccessMode#READ READ} access to the resource.
+	 * In the former case, execution of the test will occur while no other test
+	 * that uses the shared resource is being executed.
+	 * In the latter case, the test class or test method
+	 * may be executed concurrently with other tests
+	 * that also require {@code READ} access but not at the same time
+	 * as any other test that requires {@code READ_WRITE} access.
+	 *
+	 * <p>This guarantee extends to lifecycle methods of a test class or method.
+	 * For example, if a {@code Lock} is added to a test method
+	 * then the "lock" will be acquired before any
+	 * {@link BeforeEach @BeforeEach} methods are executed and released after all
+	 * {@link AfterEach @AfterEach} methods have been executed.
```

Comment:
I don't think we need to repeat all that here, in particular since we link to the Javadoc of `@ResourceLock` in the `@apiNote` below.

```suggestion
```